temperature/routers.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from temperature import crud, schemas
+from database import AsyncSessionLocal
+
+router_temp = APIRouter()
+
+
+async def get_db():
+    async with AsyncSessionLocal() as db:
+        yield db
+
+
+@router_temp.get("/", response_model=list[schemas.Temperature])
+async def read_city(db: AsyncSession = Depends(get_db)):
+    return await crud.get_temp(db=db)
+
+
+@router_temp.get("/{city_id}/", response_model=schemas.Temperature)
+async def read_single_city(city_id: int, db: AsyncSession = Depends(get_db)):
```

Comment:
The function name `read_single_city` is misleading as it suggests retrieving a city rather than a temperature record. Consider renaming it to `read_single_temperature` or similar to accurately describe its functionality.